ethstorage/archiver/guide-devnet.md

```diff
@@ -0,0 +1,366 @@
+forge create src/BatchInbox.sol:BatchInbox  \
+--constructor-args $ES_CONTRACT \
+--private-key $PRIVATE_KEY \
+--rpc-url http://localhost:8545
+
+Deployer: 0xDe3829A23DF1479438622a08a116E8Eb3f620BB5
+Deployed to: 0xb860F42DAeD06Cf3dC9C3b4B8A287523BbdB2B1e
+Transaction hash: 0x99f6788e90004a68e67fa2848e47f7592ffb38aaff31b1738bcc163d806a00a5
+```
+
+Make sure to save the deployed contract address for future use. For example:
+
+```hash
+export BATCH_INBOX=0xb860F42DAeD06Cf3dC9C3b4B8A287523BbdB2B1e  # replace with the actual address
+```
+
+Do not forget to fund the batcher in the BatchInbox account, where `0x3C44CdDdB6a900fa2b585dd299e03d12FA4293BC` is the batcher address used in the devnet:
+```bash
+cast send $BATCH_INBOX "deposit(address)" 0x3C44CdDdB6a900fa2b585dd299e03d12FA4293BC --value 100ether --private-key $PRIVATE_KEY
+```
+
+Finally, navigate to the Optimism monorepo and locate `batchInboxAddress` in the file `packages/contracts-bedrock/deploy-config/devnetL1.json`:
+
+```json
+  "batchInboxAddress": "0xff00000000000000000000000000000000000901",
+```
+Update the value of `batchInboxAddress` with the address of the contract you just deployed. 
+
+Now, navigate to the parent directory in preparation for the next steps.
+
+### Running a Proxy of L1 Beacon
+
+For the convenience of testing, you will start a proxy for the Beacon API with a shorter blob retention period.
+
+First retrieve the beacon genesis time for later use:
+```bash
+curl -s http://localhost:5052/eth/v1/beacon/genesis | jq -r '.data.genesis_time'
+
+1732529739
+
+export GENESIS_TIME=1732529739 // replace with the actual timestamp
+```
+
+The following commands start a proxy to Beacon API with a shorter blobs retension period:
+```bash
+git clone https://github.com/ethstorage/beacon-api-wrapper.git
+cd beacon-api-wrapper
+go run cmd/main.go -b http://localhost:5052 -p 3602 -g $GENESIS_TIME -r 1200
+```
+This setup allows you to test archive service effectively. 
+For blob requests, if the request is within the latest 100 slots, the proxy will retrieve blobs from `http://localhost:5052`. For requests older than that, it will return an empty list.
+
+## Running EthStorage Node
+
+
+### Installation
+
+To set up the es-node, first clone the repository and build it:
+```bash
+git clone https://github.com/ethstorage/es-node.git
+cd es-node
+git checkout v0.1.16
+make
+```
+
+### Initialization
+
+Initialize es-node:
+```bash
+./init-rpc.sh \
+--l1.rpc http://localhost:8545 \
+--storage.l1contract $ES_CONTRACT
+```
+
+### Running ES Node in Archiver Mode
+
+Finally, run the es-node:
+
+```bash
+./run-rpc.sh \
+--storage.l1contract $ES_CONTRACT \
+--l1.rpc http://localhost:8545 \
+--l1.beacon http://localhost:5052 \
+--l1.beacon-based-time $GENESIS_TIME \
+--l1.beacon-based-slot 0 \
+--p2p.listen.udp 30375 \
+--p2p.listen.tcp 9733 \
+--rpc.port 9745 \
+--archiver.port 6678 \
+--archiver.enabled 
+```
+
+Now, navigate to the parent directory in preparation for the next steps.
```

Comment:
I suggest to add some comment for this command: how it works / what exactly does it store , though I guess it stores all blobs of blob tx targeted to `$ES_CONTRACT`.

Reply:
The archiver service itself is unaware of dApps. Isn't it?

Reply:
I mean for the user of archiver, at first glance he may think an archiver can be shared by multiple dapps.